src/Fixer/ControlStructure/SimplifiedIfReturnFixer.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\ControlStructure;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Filippo Tessarotto <zoeslam@gmail.com>
+ */
+final class SimplifiedIfReturnFixer extends AbstractFixer
+{
+    private $sequences = [
+        [
+            'isNegative' => false,
+            'sequence' => [
+                ')',
+                '{', [T_RETURN], [T_STRING, 'true'], ';', '}',
+                [T_RETURN], [T_STRING, 'false'], ';',
+            ],
+        ],
+        [
+            'isNegative' => true,
+            'sequence' => [
+                ')',
+                '{', [T_RETURN], [T_STRING, 'false'], ';', '}',
+                [T_RETURN], [T_STRING, 'true'], ';',
+            ],
+        ],
+        [
+            'isNegative' => false,
+            'sequence' => [
+                ')',
+                [T_RETURN], [T_STRING, 'true'], ';',
+                [T_RETURN], [T_STRING, 'false'], ';',
+            ],
+        ],
+        [
+            'isNegative' => true,
+            'sequence' => [
+                ')',
+                [T_RETURN], [T_STRING, 'false'], ';',
+                [T_RETURN], [T_STRING, 'true'], ';',
+            ],
+        ],
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Simplify `if` control structures that return the boolean result of their condition.',
+            [new CodeSample("<?php\nif (\$foo) { return true; } return false;\n")]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // should be run before NoMultilineWhitespaceBeforeSemicolonsFixer, NoSinglelineWhitespaceBeforeSemicolonsFixer.
+        return 1;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAllTokenKindsFound([T_IF, T_RETURN, T_STRING]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $foundIndexes = $tokens->findGivenKind([T_IF, T_ELSEIF]);
```

Comment:
Currently the fixer scans all tokens for all `T_IF` / `T_ELSEIF` and than iterates the reverse of that. For each match it search for the end of the body of that if/else match and than goes back to the start of the body and scans again within the body up to four times for the sequences. 

Maybe we can start at the end of the tokens and iterate back using int based indexes and than check for `T_IF` / `T_ELSEIF` ? And than if there is a match check if there is a candidate with early detection there isn't any?

Reply:
I don't get your last sentence :confused: 
Yes performance could be improved, but I'd prefer much more to let `Tokens` class provide performing operations and keep fixers' code readable and understandable.

Reply:
Nearly all fixers just iterate over the tokens and check for whatever kind of token they are looking for, nearly none scan all the tokens first and than go fixing. So for readable and understandable, I think it best to not introduce another method.

Reply:
I'm sorry, so is `findGivenKind` call the performance degradator you are talking about? I thought is is the least of the problems...

Reply:
feel free to not look into it, I'm just testing the code and share my findings, it is your PR of course

Reply:
I admit I lost myself in what you are asking me. I want to be clear that it was never my intention to go against this library conventions or best practices.
So, given that I didn't understand much, I'll change the `findGivenKind` to a simple loop through the tokens; any other change request will be applied as well, as long as I understand them :stuck_out_tongue_closed_eyes: